tensorflow_recommenders_addons/dynamic_embedding/python/ops/cuckoo_hashtable_ops.py

```diff
@@ -316,12 +313,12 @@ def __init__(self, table, name, full_name=""):
       ]
       # pylint: disable=protected-access
       super(CuckooHashTable._Saveable, self).__init__(table, specs, name)
-      self.full_name = full_name
+      self._restore_name = table._name
 
     def restore(self, restored_tensors, restored_shapes, name=None):
       del restored_shapes  # unused
       # pylint: disable=protected-access
-      with ops.name_scope(name, "%s_table_restore" % self.name):
+      with ops.name_scope(name, "%s_table_restore" % self._restore_name):
```

Comment:
Is there backward compatibility problem? After naming policy, New version TFRA can restore checkpoint created by old version ,if not maybe we had to inform user in some way..

Reply:
It does break some compatibility to the backward versions. In graph mode, the `Saver` only saves the `CuckooHashTable` and will not save the `Variable`, since `Variable` only holds a list of tables. When the `var_list` should be `var_list=var.tables`, if it is explicitly set. And in eager mode, saved_model or checkpoint can automatically track the tables from `Variable`. And also the naming will be slightly different from earlier versions.

Reply:
Hi @Mr-Nineteen ,can you help review here? thanks!

Reply:
Currently, it is compatible with the older version's checkpoint and saved_model.